lib/Controller/CallController.php

```diff
@@ -104,9 +104,13 @@ public function getPeersForCall(): DataResponse {
 			if ($this->getAPIVersion() >= 3) {
 				$displayName = $participant->getAttendee()->getActorId();
 				if ($participant->getAttendee()->getActorType() === Attendee::ACTOR_USERS) {
-					$user = $this->userManager->get($participant->getAttendee()->getActorId());
-					if ($user instanceof IUser) {
-						$displayName = $user->getDisplayName();
+					if ($participant->getAttendee()->getDisplayName()) {
+						$displayName = $participant->getAttendee()->getDisplayName();
+					} else {
+						$user = $this->userManager->get($participant->getAttendee()->getActorId());
+						if ($user instanceof IUser) {
+							$displayName = $user->getDisplayName();
```

Comment:
shouldn't we also store the display name in the attendee table also after retrieving it here, if the purpose is for caching it there ?

otherwise it looks like we'd only store it there if the user changes the display name, which is probably seldom.
unless I missed some code path that is not directly visible in this diff ?

Reply:
same comment for the other similar block

if that duplicated logic gets bigger we could move it into a common utility method maybe

Reply:
Let's improve step by step.
But I would really like to avoid potentially lot of update queries making this request even more heavier.

We can add a repairstep/background job or something. But future story.

Reply:
Fair enough.

So far I still don't understand when that attendee display name is getting populated. From my understanding it will almost never get any values, unless people are changing their display names while in a call ? So it seems it's almost never happening ?

Reply:
(let me actually test this, that might answer my question)

Reply:
Okay, after a test I saw that the display name is not populated in the attendees table, not even during a call.
This means that during a big call we'd still go to the user manager for display names (unless I missed something).

The only way to get an entry there is changing your display name in profile page.
So, from my understanding, the performance benefit is not achieved, at least not in this PR.

We'll then need the repair steps, background job or whatever is supposed to populate there, then.

Why not also populate the field at the time the user is getting added to the conversation ? We're anyway inserting new entries there.

Reply:
LDAP automatically triggers an update periodically as far as i know. But then again, inserting it via the hook is step 1, initial population for existing entries is step 2